Directory.Build.targets

```diff
@@ -12,6 +12,21 @@
   <Import Project="$(RepositoryEngineeringDir)generators.targets" />
   <Import Project="$(RepositoryEngineeringDir)python.targets" />
 
+  <!--
+  When .NET gets built from source, make the SDK aware there are bootstrap packages
+  for Microsoft.NETCore.App.Runtime.<rid> and Microsoft.NETCore.App.Crossgen2.<rid>.
+  -->
+  <ItemGroup Condition=" '$(DotNetBuildFromSource)' == 'true' " >
+    <KnownFrameworkReference Update="Microsoft.NETCore.App">
+      <RuntimePackRuntimeIdentifiers>$(PackageRID)</RuntimePackRuntimeIdentifiers>
+    </KnownFrameworkReference>
+    <KnownCrossgen2Pack Update="Microsoft.NETCore.App.Crossgen2">
+      <Crossgen2RuntimeIdentifiers>$(PackageRID)</Crossgen2RuntimeIdentifiers>
+    </KnownCrossgen2Pack>
+    <!-- Avoid references to Microsoft.AspNetCore.App.Runtime.<rid> -->
+    <KnownFrameworkReference Remove="Microsoft.AspNetCore.App" />
```

Comment:
Don't we also need to remove the WindowsDesktop KnownFrameworkReference?

Reply:
I don't see these with the source-build prebuilts, so I assume there is no reference made by the SDK just by adding `RuntimeIdentifier`.

Reply:
I do see it in a sample console app when specifying `SelfContained=true;RuntimeIdentifier=win10-x64`:

<img width="589" alt="image" src="https://user-images.githubusercontent.com/7412651/192819299-5a2ce706-c17f-47da-9bbc-3592a6cb96b8.png">

Reply:
It's not being referenced during source-build.
Should I add it here anyway?

Reply:
Sorry for not responding. It's weird that we don't see such prebuilts even though we set a runtime identifier. Maybe `ProcessFrameworkReferences` doesn't run for the projects for which we specify a RID?

Reply:
I don't know. Maybe the reference is added only when the host/target/sdk is Windows.